scrapy_poet/backend.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+import attr
+import scrapy
+from scrapy.utils.defer import maybe_deferred_to_future
+from web_poet import HttpRequest
+from web_poet.exceptions import (
+    HttpError,
+    HttpRequestError,
+    RequestBackendError,
+)
+
+from scrapy_poet.utils import scrapy_response_to_http_response
+
+logger = logging.getLogger(__name__)
+
+
+def create_scrapy_backend(backend):
+    async def scrapy_backend(request: HttpRequest):
+        if not isinstance(request, HttpRequest):
+            raise TypeError(
+                f"The request should be 'web_poet.HttpRequest' but received "
+                f"one of type: '{type(request)}'."
+            )
+
+        try:
+            request = scrapy.Request(**attr.asdict(request), dont_filter=True)
+        except TypeError:
+            raise RequestBackendError(
+                f"The given Request isn't compatible with `scrapy.Request`. "
+                f"Ensure that it doesn't contain extra fields: {request}"
```

Comment:
I wonder how can users act on this suggestion

Reply:
Should we expose a way (e.g. a setting) for users to provide their custom implementation of web poet request to Scrapy request conversion?

Or should we maybe reconsider how we handle this? For example, ignore extra fields, possibly with a warning.

Reply:
I'm not sure when could the user get this message tbh. web_poet.HttpRequest is a well-defined class, scrapy.Request is a well-defined class, why would there be more than one way to convert between those?

Reply:
I guess we can assume there will be no HttpRequest subclasses requiring special conversion logic. We can always revisit in the future if someone comes up with a valid use case.